### ISSUE TITLE
Is there any plan to port this project to ROS2?

### ISSUE THREAD

**Issue by xhuan28** [ANYbotics/rqt_multiplot_plugin#38]
Is there any plan to port this project to ROS2?

**Comment by yossioo** [comment]
@kralf 
Maybe some guides where to start for those who dare to try? :)

**Comment by AmaurX** [comment]
I would also be very interested for a ROS2 Foxy version. I'll like to try but a guide to which files are specific to ROS and which are not would be nice.
Also, one big issue will definitely be the issue of QoS compatibility between publishers and subscribers

**Comment by GuihVicentini** [comment]
Any update on this? I am also interested on running multiplot on ROS2. Did anyone found another solution or a guideline to adapt this solution?

**Comment by remod** [comment]
Hello everyone,

we do not have plans at the moment to port this package to ROS2. If anyone wants to contribute, I am happy to create a `ros2` branch!

**Comment by GuihVicentini** [comment]
Hi @remod I kind of have interest on this, but I am clueless on what are exactly the ROS1 dependencies on this project. Could you give me a starting point?

**Comment by remod** [comment]
Hi @GuihVicentini 

I am unsure whether I understood the question correctly - does the [migration guide] to ROS2 help?

**Comment by abrzozowski** [comment]
Hi @remod,

First overall, @kralf, @remod and others, kudos for the fantastic project!

It would be super to have this plugin in ROS2. Were there some steps taken in this direction?
If not, I can take care of it. I will be doing it in my free time, so it probably will take a while to do this porting, but I hope not longer than to end of this year :P

**Comment by remod** [comment]
Hi @abrzozowski 

That would be great, thank you! We're still using ROS1 and therefore cannot assist with the implementation, but we can help with the review! How would you like to proceed? Create a `ros2` branch?

**Comment by abrzozowski** [comment]
Hi @remod,

Great! I will create a fork of this repo, and I will be porting on this fork. When everything works, I will create a PR to this main repo (https://github.com/ANYbotics/rqt_multiplot_plugin). Then we'll do the review. Is it a good plan?

I think that creating a `ros2` branch in the main repo is an excellent idea because it can show others that there is someone who works on porting.

**Comment by remod** [comment]
Yes that sounds great!

One input from my side: ROS2 brings along some code formatting tools. To ensure that potential bugfixes are compatible with both versions of `rqt_multiplot_plugin`, ROS1 and ROS2, I suggest that we do not reformat code on the ROS2 branch - is this fine for you?

**Comment by abrzozowski** [comment]
Ok, I thought about it. I will be changing as minimally as possible. This approach should help with future merging `ros1`-releases into `ros2` as well

**Comment by abrzozowski** [comment]
I've taken a look at what is a work scope. There is one more package to port except this, a https://github.com/ANYbotics/variant. The [rqt_multiplot_plugin](https://github.com/ANYbotics/rqt_multiplot_plugin) is dependent on the [variant](https://github.com/ANYbotics/variant)

**Comment by remod** [comment]
Yes exactly. In case you are not aware, the `variant` package does some ROS message definition parsing. Porting it to ROS2 might involve a bit more work than usually when porting a node.

**Comment by nlamprian** [comment]
Is there any ongoing activity on this topic?

**Comment by remod** [comment]
Hi @nlamprian !

I personally am not aware of any activity.

**Comment by nlamprian** [comment]
OK, I'll see if I can get involved. This tool deserves to stay alive!

**Comment by abrzozowski** [comment]
Hi, guys. Unfortunately, from my side, the porting didn't really move forward. The variant also still needs to be ported. But I still hold onto the hope that I'll get it done this year.

**Comment by amburkoff** [comment]
I really liked using rqt_multiplot on ros1. Any news on porting to ros2?

**Comment by abrzozowski** [comment]
Unfortunately, I haven't found the time so far to do it. So, officially, I'm saying that I'm not currently dealing with it :(

**Comment by samuelba** [comment]
working ROS2 port https://github.com/samuelba/rqt_multiplot_plugin, including some new features. It is not yet released to ROS PPA, but I'm working on it.
